rpi_5 fails to build samples/tests with UART

**Describe the bug**
rpi_5 fails to build samples/tests with UART

**To Reproduce**
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -DBOARD=rpi_5  ../samples/subsys/shell/shell_module/ -GNinja
3. ninja
4. See error

**Expected behavior**
No build errors

**Impact**
CI failures when this board is built for any of the affected samples

**Logs and console output**
https://github.com/zephyrproject-rtos/zephyr/actions/runs/9110213175/job/25045490905?pr=72860#step:12:347
```
/__w/zephyr/zephyr/twister-out/rpi_5/samples/posix/env/sample.posix.env/zephyr/include/generated/devicetree_generated.h:1419:31: error: implicit declaration of function 'DT_N_S_soc_S_serial_107d001000_P_interrupt_names_FOREACH_PROP_ELEM'; did you mean 'DT_N_S_soc_S_serial_107d001000_P_status_FOREACH_PROP_ELEM'? [-Werror=implicit-function-declaration]
 1419 | #define DT_N_INST_0_arm_pl011 DT_N_S_soc_S_serial_107d001000
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
...
/__w/zephyr/zephyr/include/zephyr/devicetree.h:4338:30: note: in expansion of macro 'DT_DRV_INST'
 4338 |         DT_FOREACH_PROP_ELEM(DT_DRV_INST(inst), prop, fn)
      |                              ^~~~~~~~~~~
/__w/zephyr/zephyr/drivers/serial/uart_pl011.c:612:17: note: in expansion of macro 'DT_INST_FOREACH_PROP_ELEM'
  612 |                 DT_INST_FOREACH_PROP_ELEM(n, interrupt_names,                           \
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~
/__w/zephyr/zephyr/drivers/serial/uart_pl011.c:636:9: note: in expansion of macro 'PL011_CONFIG_PORT'
  636 |         PL011_CONFIG_PORT(n)                                    \
      |         ^~~~~~~~~~~~~~~~~
``` 

**Environment (please complete the following information):**
 - OS: Linux (CI , Ubuntu 22.04)
 - Toolchain: Zephyr SDK 0.16.5[-1]
 - Commit SHA or Version used: Main as of now c8cd57fe9ef4547b2d92d4733af57acdeb8e08e8

**Additional info**
Need for the interrupt-names property was added in #57923
Support for serial in this platform was added in 63692c1349f767c5c78ad30fb4b024c80cb61329 together with the platform (#70538) where it was already broken.

--- a/dts/arm64/broadcom/bcm2712.dtsi
+++ b/dts/arm64/broadcom/bcm2712.dtsi
@@ -73,6 +73,7 @@
 			reg = <0x10 0x7d001000 0x200>;
 			interrupts = <GIC_SPI 121 IRQ_TYPE_LEVEL
 				      IRQ_DEFAULT_PRIORITY>;
+			interrupt-names = "irq_121";
 			clocks = <&clk_uart>;
 			status = "disabled";
 		};

